key.go

```diff
@@ -420,3 +420,31 @@ func GenerateRSAKeyWithExponent(bits int, exponent int) (PrivateKey, error) {
 	})
 	return p, nil
 }
+
+// DeriveKey generates a key and IV from given password and salt using openssl EVP_BytesToKey()
+func DeriveKey(cipher *Cipher, digest *Digest, salt []byte, password []byte,
+	iterations int) (key, iv []byte, err error) {
+	key = make([]byte, cipher.KeySize())
+	iv = make([]byte, cipher.IVSize())
+
+	var saltPtr, ivPtr, passwordPtr, keyPtr *C.uchar
+	if len(salt) != 0 {
```

Comment:
so reading some openssl docs (https://wiki.openssl.org/index.php/Manual:EVP_BytesToKey(3)) this field should be either 8 bytes or null. can we have the function return an error if salt is passed and not 8 bytes?